fake_switches/juniper/juniper_netconf_datastore.py

```diff
@@ -307,6 +311,17 @@ def parse_vlans(self, conf, etree_conf):
 
         return handled_elements
 
+    def parse_native_vlan(self, interface_node, port):
+        if len(interface_node.xpath("unit/family/ethernet-switching/native-vlan-id")) == 1 and interface_node.xpath("unit/family/ethernet-switching/native-vlan-id")[0].text is not None:
+            port_attributes = first(interface_node.xpath("unit/family/ethernet-switching"))
+            return resolve_new_value(port_attributes, "native-vlan-id", port.trunk_native_vlan,
+                              transformer=int)
+        return None
+
+    def apply_trunk_native_vlan_patch(self, interface_data, port):
```

Comment:
Maybe it's a bad use of the term. But essentially, when it's qfx switch, I patch the tree generated by the default juniper code to qfx tree map.